src/plugins/console/public/lib/mappings/mappings.ts

```diff
@@ -29,18 +29,63 @@
  */
 
 import _ from 'lodash';
+import { HttpResponse, HttpSetup } from 'opensearch-dashboards/public';
 import * as opensearch from '../opensearch/opensearch';
+import { Settings } from '../../services';
+
+export interface Field {
+  name: string;
+  type: string;
+}
+
+interface FieldMapping {
+  enabled?: boolean;
+  path?: string;
+  properties?: Properties;
+  type?: string;
+  index_name?: string;
+  fields?: Record<string, FieldMapping>;
+  index?: string;
+}
+
+type Properties = Record<string, FieldMapping>;
+
+interface IndexMapping {
+  [index: string]: {
+    [mapping: string]: FieldMapping | Properties;
+  };
+}
+
+interface IndexMappingOld {
+  [index: string]: {
+    mappings: {
+      properties: Properties;
+    };
+  };
+}
+
+interface Aliases {
+  [aliasName: string]: Record<string, unknown>;
+}
+
+interface IndexAliases {
+  [indexName: string]: {
+    aliases: Aliases;
+  };
+}
 
 // NOTE: If this value ever changes to be a few seconds or less, it might introduce flakiness
 // due to timing issues in our app.js tests.
 const POLL_INTERVAL = 60000;
-let pollTimeoutId;
+let pollTimeoutId: NodeJS.Timeout | null;
 
-let perIndexTypes = {};
-let perAliasIndices = {};
-let templates = [];
+let perIndexTypes: { [index: string]: { [type: string]: Field[] } } = {};
+let perAliasIndices: { [alias: string]: string[] } = {};
+let templates: string[] = [];
 
-export function expandAliases(indicesOrAliases) {
+export function expandAliases(
+  indicesOrAliases: string | string[] | null | undefined
```

Comment:
Do we want to add a type for `string | string[] | null` so it is more clear and less duplicate?  Maybe something like `type IndicesOrAliases = string | string[] | null;`